### PR TITLE
Adding the option to disable the adventure game mode on archived worlds

### PR DESCRIPTION
**Pull request by paaulhier**
-----

- Adding an option into the config.yml file to disable the adventure game mode on archived worlds

See https://github.com/thomasmny/BuildSystem/issues/286

**Please note: we will close your Pull Request without comment if you do not check the box above and provide ALL
requested information.**

### CODE REVIEW

**Comment by thomasmny** [comment]
Looks good so far. 
However, we could also give players the option to set which gamemode players should be put into as well:
- `achive-gamemode` (defaults to adventure)
- `archive-set-gamemode` (defaults to true)

**Comment by paaulhier** [comment]
That's actually a good suggestion. Will implement it today afternoon

**Comment by paaulhier** [comment]
Implemented it now, may you check again, please?

File: buildsystem-core/src/main/java/de/eintosti/buildsystem/listener/PlayerChangedWorldListener.java [hunk 1]
```diff
@@ -189,7 +189,10 @@ private void checkWorldStatus(Player player) {
     }
 
     private void setSpectatorMode(Player player) {
-        player.setGameMode(GameMode.ADVENTURE);
+        // Checking if the game mode should be set to adventure mode on archive worlds
```
Comment (thomasmny) [inline buildsystem-core/src/main/java/de/eintosti/buildsystem/listener/PlayerChangedWorldListener.java:192]: You can remove the comment, code should be self-explanatory

**Review by thomasmny — CHANGES_REQUESTED** [review]
All in all very good, just a few nitpicks

**Pull request merged**